instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/OpenTelemetryKafkaMetrics.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  /**
+   * Reset for test by reseting the {@link #meter} to {@code null} and closing all registered
+   * instruments.
+   */
+  static void resetForTest() {
+    meter = null;
+    for (Iterator<Map.Entry<RegisteredInstrument, AutoCloseable>> it =
+            instrumentMap.entrySet().iterator();
+        it.hasNext(); ) {
+      try {
+        it.next().getValue().close();
+        it.remove();
+      } catch (Exception e) {
+        throw new IllegalStateException("Error occurred closing instrument", e);
+      }
+    }
+  }
+
+  /**
+   * Print a table mapping kafka metrics to equivalent OpenTelemetry metrics, in markdown format.
+   */
+  static void printMappingTable() {
+    StringBuilder sb = new StringBuilder();
+    // Append table headers
+    sb.append(
+            "| # | Metric Group | Metric Name | Attribute Keys | Instrument Name | Instrument Description | Instrument Type |")
+        .append(lineSeparator())
+        .append(
+            "|---|--------------|-------------|----------------|-----------------|------------------------|-----------------|")
+        .append(lineSeparator());
+    Map<String, List<KafkaMetricId>> kafkaMetricsByGroup =
+        seenMetrics.stream().collect(groupingBy(KafkaMetricId::getGroup));
+    int count = 1;
+    // Iterate through groups in alpha order
+    for (String group : kafkaMetricsByGroup.keySet().stream().sorted().collect(toList())) {
+      List<KafkaMetricId> kafkaMetricIds =
+          kafkaMetricsByGroup.get(group).stream()
+              .sorted(comparing(KafkaMetricId::getName))
+              .collect(toList());
+      // Iterate through metrics in alpha order by name
+      for (KafkaMetricId kafkaMetricId : kafkaMetricIds) {
+        Optional<InstrumentDescriptor> descriptor =
+            Optional.ofNullable(KafkaMetricRegistry.getInstrumentDescriptor(kafkaMetricId));
+        // Append table row
+        sb.append(
+            String.format(
+                "| %s | %s | %s | %s | %s | %s | %s |%n",
+                count,
+                group,
+                kafkaMetricId.getName(),
+                String.join(",", kafkaMetricId.getAttributeKeys()),
+                descriptor.map(InstrumentDescriptor::getName).orElse(""),
+                descriptor.map(InstrumentDescriptor::getDescription).orElse(""),
+                descriptor.map(InstrumentDescriptor::getInstrumentType).orElse("")));
+        count++;
+      }
+    }
+    logger.log(Level.INFO, "Mapping table" + System.lineSeparator() + sb);
+  }
+
+  @Override
+  public void init(List<KafkaMetric> metrics) {
+    metrics.forEach(this::metricChange);
+  }
+
+  @Override
+  public void metricChange(KafkaMetric metric) {
+    KafkaMetricId kafkaMetricId = KafkaMetricId.create(metric);
+    seenMetrics.add(KafkaMetricId.create(metric));
+    Meter currentMeter = meter;
+    if (currentMeter == null) {
+      logger.log(Level.FINEST, "Metric changed but meter not set: " + kafkaMetricId);
```

Comment:
WDYT about using placeholders? (Especially for fine/finest logs)
```suggestion
      logger.log(Level.FINEST, "Metric changed but meter not set: {0}", kafkaMetricId);
```